File: boj/1684.py
from math import gcd
# 하... python gcd


N=int(input())

# def gcd(a, b):
#     c = max(a, b)
#     d = min(a, b)

#     while True:
#         if c % d != 0:
#             temp = c % d
#             c = d
#             d = temp
#         else:
#             break
    
#     return d
    
li=list(map(int, input().split()))
new_li = []
li.sort()

for i in range(N-1):
    new_li.append(li[i+1] - li[i])

new_li = [x for x in new_li if x != 0]

#ans=new_li[0]

# python에는 gcd가 있다.. python의 그의 한계는 어디까지인가?
# 파이썬 펀치! 파이썬 펀치! 파이썬 펀치! 파이썬 펀치! 그는 신이야!


ans = gcd(*new_li)

# for i in new_li:
#     ans = gcd(i, ans)
print(ans)

File: boj/1684.java
import java.io.*;
import java.math.*;
import java.util.*;

public class Main {

    public static void temp(Stack<Character> left, Stack<Character> right, char[] a) {
        for (int i = 0; i < a.length; i++) {
            left.push(a[i]);
        }
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        br.readLine();
        String[] temp = br.readLine().split(" ");
        ArrayList<Long> real = new ArrayList<>();

        Long[] r = Arrays.stream(temp)
                .map(Long::new)
                .sorted()
                .toArray(Long[]::new);

        if (r.length == 1) {
            bw.write(r[0] + "\n");
        } else {

            for (int i = 1; i < r.length; i++) {
                real.add(r[i] - r[i - 1]);
            }

            BigInteger[] ans = real
                    .stream()
                    .filter(x -> x != 0)
                    .map(BigInteger::valueOf)
                    .toArray(BigInteger[]::new);

            BigInteger gcd = ans[0];

            for (int i = 0; i < ans.length; i++) {
                gcd = gcd.gcd(ans[i]);
            }

            bw.write(gcd + "\n");
        }


        bw.close();
    }
}